src/codechandler/AVCCodecHandler.cpp

```diff
@@ -0,0 +1,203 @@
+#include "AVCCodecHandler.h"
+
+AVCCodecHandler::AVCCodecHandler(AP4_SampleDescription* sd)
+  : CodecHandler(sd), m_countPictureSetIds(0), m_needSliceInfo(false)
+{
+  unsigned int width(0), height(0);
+  if (AP4_VideoSampleDescription* video_sample_description =
+          AP4_DYNAMIC_CAST(AP4_VideoSampleDescription, m_sampleDescription))
+  {
+    width = video_sample_description->GetWidth();
+    height = video_sample_description->GetHeight();
+  }
+  if (AP4_AvcSampleDescription* avc =
+          AP4_DYNAMIC_CAST(AP4_AvcSampleDescription, m_sampleDescription))
+  {
+    m_extraData.SetData(avc->GetRawBytes().GetData(), avc->GetRawBytes().GetDataSize());
+    m_countPictureSetIds = avc->GetPictureParameters().ItemCount();
+    m_naluLengthSize = avc->GetNaluLengthSize();
+    m_needSliceInfo = (m_countPictureSetIds > 1 || !width || !height);
```

Comment:
avoid handle integer variables as booleans,
maybe:
```suggestion
    m_needSliceInfo = (m_countPictureSetIds > 1 || width > 0 || height > 0);
```